components/sponsored_benefits/app/models/sponsored_benefits/organizations/plan_design_proposal.rb

```diff
@@ -4,22 +4,67 @@ class PlanDesignProposal
 
       include Mongoid::Document
       include Mongoid::Timestamps
+      include AASM
 
       embedded_in :plan_design_organization, class_name: "SponsoredBenefits::Organizations::PlanDesignOrganization"
 
       field :title, type: String
-      field :claim_date, type: Date
       field :submitted_date, type: Date
 
+      field :claim_code, type: String
+      field :claim_date, type: Date
+      field :published_on, type: Date
+      field :aasm_state, type: String
+
       embeds_one :profile, class_name: "SponsoredBenefits::Organizations::AcaShopCcaEmployerProfile"
 
+      validates_uniqueness_of :claim_code, :case_sensitive => false, :allow_nil => true
+
       scope :datatable_search, ->(query) { self.where({"$or" => ([{"title" => Regexp.compile(Regexp.escape(query), true)}])}) }
 
 
       def self.find(id)
         organization = SponsoredBenefits::Organizations::PlanDesignOrganization.where("plan_design_proposals._id" => BSON::ObjectId.from_string(id)).first
         organization.plan_design_proposals.detect{|proposal| proposal.id == BSON::ObjectId.from_string(id)}
       end
+
+      def can_quote_be_published?
+        # FIXME, do we have any validations, if yes what are they
```

Comment:
bring this up tomorrow during the call so we can confirm there's nothing we're missing.

PlanDesignOrganization already requires a sic code, and I think the office_location validates presence of zip / county. So maybe we just need to confirm there's an assigned reference_plan_id to this proposal?